### PR TITLE
pkg/boot: shared bootloader logic.

### PR DESCRIPTION
**Pull request by hugelgupf**
@Natolumin and @insomniacslk 

the intent is for pkg/boot to be used in pxeboot and disk boot and netboot21. The design is somewhat copied from #509 :)

### CODE REVIEW

File: pkg/boot/boot.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+)
+
+// OSImage represents a bootable OS package.
+type OSImage interface {
+	ExecutionInfo(log *log.Logger)
+
+	// Execute kexec's the OS image: it loads the OS image into memory and
+	// jumps to the kernel's entry point.
+	//
+	// If dryRun is true, Execute should write the OS image to tmpfs and
```
Comment (Natolumin) [inline pkg/boot/boot.go]: dryRun is no more, but this comment was not updated

File: pkg/boot/linux.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+		li.Cmdline = string(b)
+	}
+
+	if initrd, ok := a.Files["modules/initrd/content"]; ok {
+		li.Initrd = initrd
+	}
+	return li, nil
+}
+
+func NewLinuxImage(kernel, initrd io.ReaderAt, cmdline string) *LinuxImage {
```
Comment (Natolumin) [inline pkg/boot/linux.go]: I'm not sure this function is really useful, since LinuxImage is exported and has all its fields exported

Comment (hugelgupf) [reply pkg/boot/linux.go]: Good point, dropped it.

File: pkg/boot/boot.go (hunk 1 above)
Comment (hugelgupf) [reply pkg/boot/boot.go]: Done.

**Comment by hugelgupf** [comment]
If you're looking at this again, please review #694 first. Otherwise, lots of changes still incoming.

**Comment by hugelgupf** [comment]
cc @samee

File: pkg/boot/measurement.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
+package boot
```
Comment (rminnich) [inline pkg/boot/measurement.go]: this seems so generally useful (measuringreader) I wonder why it's not in the cpio package.

Comment (hugelgupf) [reply pkg/boot/measurement.go]: Right now, the measuring reader is specific to the netboot21 format, with a signature and signature_algo file in the cpio. Maybe we can make this more general at some point later.

**Pull request merged**